App.Currencies/BusinessLogic/CurrencyRateManager.cs

```diff
@@ -22,7 +23,10 @@ public void AddCurrencyRate(CurrencyRate currencyRate)
 
         public CurrencyRate GetCurrencyRate(DateTime date, string currencyName)
         {
-            return this._currencyRateRepository.GetCurrencyRate(date, currencyName);
+            var result = this._currencyRateRepository.GetCurrencyRate(date, currencyName);
+            if (result == null)
+                throw new NotFoundException("There is no such currency");
```

Comment:
you fetch currency rate but in case of null you say "currency not found". it is a bit confusing